OnlineFlightSearchAPI/Services/FlightServices/SearchFlightService.cs

```diff
@@ -0,0 +1,127 @@
+            FlightDetail flight6 = new FlightDetail("CM 2006", "LTN", "IAD", "2018-12-05 08:00", "12:35", 45, 20);
+            FlightDetail flight7 = new FlightDetail("CM 2007", "IAD", "LTN", "2018-12-05 06:00", "12:30", 40, 33);
+            FlightDetail flight8 = new FlightDetail("CM 2008", "LTN", "BUD", "2018-12-05 08:00", "2:05", 78, 0);
+            FlightDetail flight9 = new FlightDetail("CM 2009", "BUD", "LTN", "2018-12-05 10:00", "2:00", 90, 0);
+            FlightDetail flight10 = new FlightDetail("CM 2010", "IAD", "LTN", "2018-12-05 20:00", "14:35", 190, 1);
+
+            flightDetails.Add(flight1);
+            flightDetails.Add(flight2);
+            flightDetails.Add(flight3);
+            flightDetails.Add(flight4);
+            flightDetails.Add(flight5);
+            flightDetails.Add(flight6);
+            flightDetails.Add(flight7);
+            flightDetails.Add(flight8);
+            flightDetails.Add(flight9);
+            flightDetails.Add(flight10);
+        }
+
+        public List<FlightDetail> SearchFlightDetails(string startLocation, string endLocation, DateTime departureDate)
+        {
+            if(ValidateStartLocation(startLocation))
+            {
+                if(ValidateDestination(endLocation))
+                {
+                    if(ValidateDepartureDate(departureDate))
+                    {
+                        if(ValidateStartAndEndLocation(startLocation,endLocation))
+                        {
+                            return FetchFlightDetails(startLocation, endLocation, departureDate);
+                        }
+                        
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private bool ValidateStartAndEndLocation(string startLocation, string endLocation)
+        {
+            if(startLocation.Equals(endLocation))
+            {
+                throw new ValidationException(ValidationMessages.StartandEndLocationCannotBeSame);
+            }
+
+            return true;
+        }
+
+        private List<FlightDetail> FetchFlightDetails(string startLocation, string endLocation, DateTime departureDate)
+        {
+            var searchResult = flightDetails.Where(x => (x.StartLocation == startLocation) &&
+                                                        (x.Destination == endLocation) &&
+                                                        (x.DepartureDate.Date.ToString() == departureDate.Date.ToString()))
+                                                        .ToList();
+            if (searchResult.Count == 0)
+            {
+                throw new ValidationException(ValidationMessages.NoFlightsAvailable);
+            }
+
+            return searchResult;
+        }
+
+        private bool ValidateDepartureDate(DateTime departureDate)
+        {
+            if (DateTime.UtcNow > departureDate)
```

Comment:
May be create a method to equate datetime with some margin?

Reply:
This case is kept so that user shouldn't be able choose the departure date as past date.